### PR TITLE
Datacollection documentation.

### PR DESCRIPTION
**Pull request by harshjain2**

### CODE REVIEW

File: RFCs/0006-DataCollection-Protocol.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+## Motivation
+This note provides the following : 
+* Protocol details for interaction between dataCollector host process and test execution host process.
+* Data collection extensibility.
+
+## Detailed Design
+
+### Terminology
+* **IDE adapter (existing dotnet-test)**: Component that listens to messages from dotnet-test and populates the IDE with tests discovered or test results. For VS IDE this is the projectK Adapter.
```
Comment (AbhitejJohn) [inline RFCs/0006-DataCollection-Protocol.md]: Might want to remove/modify this.

File: RFCs/0006-DataCollection-Protocol.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+* **TestRunner**: The test framework specific runner that discovers/executes tests in a container for that framework. This is dotnet-test-mstest / dotnet-test-xunit / dotnet-test-nunit.
```
Comment (AbhitejJohn) [inline RFCs/0006-DataCollection-Protocol.md]: Does this still hold for the dev15 design

File: RFCs/0006-DataCollection-Protocol.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+![dotnet-test data collector protocol](Images/datacollection.png)
+
+And this is how the wire protocol looks like with custom host:
+![dotnet-test data collector protocol with custom host](Images/datacollection-custom.png)
+
+## Flow for datacollection for Run Tests:
+1. After the Adapter initiates test execution, dotnet-test checks whether data collectors are configured in test run settings. If yes, it spawns of datacollector host process.
+2. Datacollector host process then loads and initializes all the data collectors.
+3. Datacollector host process also opens another port B to be used for communicating with test execution host process.
+4. Datacolector host process then returns environment variables, if any, as specified by datacollectors and port B back to dotnet-test process.
```
Comment (AbhitejJohn) [inline RFCs/0006-DataCollection-Protocol.md]: nit: spelling.

File: RFCs/0006-DataCollection-Protocol.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+5. dotnet-test spawns of test execution host process with environment variables and passes port B.
+6. Test execution host process then connects to port B.
+7. Test execution host process send events : TestCaseStart,TestCaseEnd and TestCaseResult, on receiving them from Rocksteady adapter for every testcase.
```
Comment (AbhitejJohn) [inline RFCs/0006-DataCollection-Protocol.md]: It isn't very clear if these events are sent in sync or async.

File: RFCs/0006-DataCollection-Protocol.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+9. On completion of test execution, test-host sends DataSession.Complete.
+10. Datacollector host process returns the attachments, if any, as generated by datacollectors. 
+11. dotnet-test sends DataCollection.Terminate message to datacollector host process.
+12. Attachments returned by data collector host process is shared by loggers.
+
+## Flow for datacollection for Debug/Tests/Custom test host process:
+1. After the Adapter initiates TestExecution.GetProcessStartInfo, dotnet-test checks whether datacollectors are configured in test run settings. If yes, it spawns of datacollector host process.
+2. Datacollector host process then loads and initializes all the data collectors.
+3. Datacollector host process also opens another port B for communicating with test execution host process.
+4. Datacolector host process then returns environment variables, if any, as created by datacollectors and port B back to dotnet-test process.
```
Comment (AbhitejJohn) [inline RFCs/0006-DataCollection-Protocol.md]: nit: spelling.

File: RFCs/0006-DataCollection-Protocol.md [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+5. dotnet-test spawns of test execution host process with environment variables and passes port B.
+6. Test execution host process then connects to port B.
+7. Test execution host process send events : TestCaseStart,TestCaseEnd and TestCaseResult, on receiving them from Rocksteady adapter for every testcase.
+8. Datacollector host process responds by sending attachments, if any, as generated by datacollectors.
+9. On completion of test execution, test-host sends DataSession.Complete.
+10. Datacollector host process returns the attachments, if any, as generated by datacollectors. 
+11. dotnet-test sends DataCollection.Terminate message to datacollector host process.
+12. Attachments returned by data collector host process is shared by loggers.
+
+## Flow for datacollection for Debug/Tests/Custom test host process:
```
Comment (AbhitejJohn) [inline RFCs/0006-DataCollection-Protocol.md]: nit: Debug tests.

File: RFCs/0006-DataCollection-Protocol.md [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+2. Datacollector host process then loads and initializes all the data collectors.
+3. Datacollector host process also opens another port B for communicating with test execution host process.
+4. Datacolector host process then returns environment variables, if any, as created by datacollectors and port B back to dotnet-test process.
+5. Envionment variables and port B info is passed back to Adapter along with ProcessStartInfo of custom test host process.
+
+After the test host process is spawned, the protocol is the same as the default flow above, which is essentially from Step #6 above.
+
+## Open questions
+1. Is it required to send test case level attachments to test host process. If yes, find a use case.
+2. Is TestCase Results required by DataCollector. If yes, find a use case.
```
Comment (AbhitejJohn) [inline RFCs/0006-DataCollection-Protocol.md]: It looks like we already pass test case results to the datacollector.

File: RFCs/0006-DataCollection-Protocol.md [hunk 8]
```diff
@@ -0,0 +1,56 @@
+# 006 DataCollection Protocol
```
Comment (AbhitejJohn) [inline RFCs/0006-DataCollection-Protocol.md]: While this describes nicely the entire architecture, do we plan to have one more for folks writing an adapter? on the interfaces etc.

Comment (harshjain2) [inline RFCs/0006-DataCollection-Protocol.md]: That will follow.

File: RFCs/0006-DataCollection-Protocol.md (hunk 1 above)
Comment (harshjain2) [inline RFCs/0006-DataCollection-Protocol.md]: Done.

File: RFCs/0006-DataCollection-Protocol.md (hunk 5 above)
Comment (harshjain2) [inline RFCs/0006-DataCollection-Protocol.md]: Done.

File: RFCs/0006-DataCollection-Protocol.md (hunk 7 above)
Comment (harshjain2) [inline RFCs/0006-DataCollection-Protocol.md]: We will not pass test case results to datacollector.

File: RFCs/0006-DataCollection-Protocol.md (hunk 6 above)
Comment (harshjain2) [inline RFCs/0006-DataCollection-Protocol.md]: Done.

File: RFCs/0006-DataCollection-Protocol.md (hunk 3 above)
Comment (harshjain2) [inline RFCs/0006-DataCollection-Protocol.md]: Done.

**Comment by harshjain2** [comment]
Closing this PR because of mapping issues with open sourced repo.

**Pull request closed without merging**